src/tonic/auth.rs

```diff
@@ -29,6 +30,8 @@ async fn check(auth_keys: Arc<AuthKeys>, mut req: Request) -> Result<Request, St
         })?;
 
     let previous = req.extensions_mut().insert::<Access>(access);
+    req.extensions_mut()
+        .insert(InferenceToken::new(api_key.to_string()));
```

Comment:
Maybe we can `assert_debug` that the insertion returns None?
As far I understand, there should never be already an inference token on the request.
WDYT?

Reply:
The inference token should be updated per request, because requests could land to the inference service from different instances. Does it make sense?

Reply:
In that case, wouldn't each request start with a clean slate? The state we define here is not shared across requests, or is it?

Arnaud suggests a debug assertion as sanity check to ensure we do indeed start with a clean slate for every request. That's a valid suggestion to me, as accidentally mixing tokens across requests will be a nightmare.